props/saber_chris_buttons.h

```diff
@@ -0,0 +1,1036 @@
+Allows for adjustment to Push gesture length in millis needed to trigger Force Push
+Recommended range 1 ~ 10, 1 = shortest, easiest to trigger, 10 = longest. MM.
+
+Tightened click timings
+I've shortened the timeout for short and double click detection from 500ms
+to 300ms.  I think it feels more responsive this way but still gives enough
+timeout to ensure all button actions can be achieved consistently
+I've included all button timings so they can be easily tweaked to suit
+individual tastes. MM.
+
+============================================================
+============================================================
+*/
+#ifndef PROPS_SABER_SENSE_BUTTONS_H
+#define PROPS_SABER_SENSE_BUTTONS_H
+
+
+//  Used instead of BladeID and allows toggle switching 
+//  between two blade/preset arrays manually.
+#ifdef FAKE_BLADE_ID
+struct FakeBladeID {
+   static int return_value;
+   float id() { return return_value; }
+   static void toggle() {
+     return_value = NO_BLADE - return_value;
+   }
+};
+int FakeBladeID::return_value = 0;
+#undef BLADE_ID_CLASS_INTERNAL
+#define BLADE_ID_CLASS_INTERNAL FakeBladeID
+#undef BLADE_ID_CLASS
+#define BLADE_ID_CLASS FakeBladeID
+#endif
+
+
+#include "prop_base.h"
+#include "../sound/hybrid_font.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SabersensePropV31
+
+#ifndef MOTION_TIMEOUT
+#define MOTION_TIMEOUT 60 * 15 * 1000
+#endif
+
+#ifndef SA22C_SWING_ON_SPEED
+#define SA22C_SWING_ON_SPEED 250
+#endif
+
+#ifndef SA22C_LOCKUP_DELAY
+#define SA22C_LOCKUP_DELAY 200
+#endif
+
+#ifndef SA22C_FORCE_PUSH_LENGTH
+#define SA22C_FORCE_PUSH_LENGTH 5
+#endif
+
+#ifndef BUTTON_DOUBLE_CLICK_TIMEOUT
+#define BUTTON_DOUBLE_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_SHORT_CLICK_TIMEOUT
+#define BUTTON_SHORT_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_TIMEOUT
+#define BUTTON_HELD_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_MEDIUM_TIMEOUT
+#define BUTTON_HELD_MEDIUM_TIMEOUT 1000
+#endif
+
+#ifndef BUTTON_HELD_LONG_TIMEOUT
+#define BUTTON_HELD_LONG_TIMEOUT 2000
+#endif
+
+#ifdef SA22C_SWING_ON
+#define SWING_GESTURE
+#endif
+
+#ifdef SA22C_STAB_ON
+#define STAB_GESTURE
+#endif
+
+#ifdef SA22C_TWIST_ON
+#define TWIST_GESTURE
+#endif
+
+#ifdef SA22C_THRUST_ON
+#define THRUST_GESTURE
+#endif
+
+
+//  #define FORCE_PUSH_CONDITION battle_mode_
+
+EFFECT(dim);      // for EFFECT_POWERSAVE
+EFFECT(battery);  // for EFFECT_BATTERY_LEVEL
+EFFECT(bmbegin);  // for Begin Battle Mode
+EFFECT(bmend);    // for End Battle Mode
+EFFECT(vmbegin);  // for Begin Volume Menu
+EFFECT(vmend);    // for End Volume Menu
+EFFECT(volup);    // for increse volume
+EFFECT(voldown);  // for decrease volume
+EFFECT(volmin);   // for minimum volume reached
+EFFECT(volmax);   // for maximum volume reached
+EFFECT(faston);   // for EFFECT_FAST_ON
+EFFECT(blstbgn);  // for Begin Multi-Blast
+EFFECT(blstend);  // for End Multi-Blast
+EFFECT(push);     // for Force Push gesture in Battle Mode
+EFFECT(quote);    // for playing quotes
```

Comment:
If I remember correctly, this is in hybrid_font.h now...

Reply:
I had a look and I can't see this extract in hybrid_font.h. Also I tried removing this EFFECT list from the prop, but the result was that it wouldn't compile. What have I missed?

Reply:
I didn't mean the whole list, just EFFECT(quote)

Reply:
Ah sorry. 
Noting your comment would not potentially go live until OS-8, I can see what's happened now - the quote line is indeed handled in hybrid_font.h in OS-8 (GitHub Master) but not OS-7. Apologies - all my testing has been done in OS-7.14, in which case that line is needed.

I'll test everything today in OS-8, but early indications are that there are two issues between OS-7 and OS-8:

The first is the EFFECT(quote); line above which we now know about.

The second is on line 969.
OS-7 seems to want this:
    void SB_Effect(EffectType effect, float location) override {

But OS-8 wants this:
    void SB_Effect(EffectType effect, EffectLocation location) override {

Neither OS will compile with the wrong line.

I could put them both on an #ifndef define so we're at least backwards compatible with OS-7. Or are we not worried about that?

The rest of the formatting stuff I'm working through now. Will upload the (hopefully) correct version later today.

Reply:
Oh, and thanks for your patience with this - much appreciated as always.

Reply:
@Sabersense-CC Just FYI - Lots of updated spots through the OS will use EffectLocation for location now.
Backwards compatibility is not typically a thing with prop files. They will move forward and be updated if needed, but I've not seen one go backwards to work with a previous version from when they were added.
That said, I think you should be using the master version here for development.

Reply:
> ...Backwards compatibility is not typically a thing with prop files....

Thanks Brian. All noted. I had a hunch that was the case, but the main reason I've been using 7.14 for the time being is because colour change didn't seem to work right under the Master version, despite the fact I haven't made any changes to any colour change stuff. Hence I wanted to get everything working on a stable system first, and then make the necessary tweaks to make it work with OS-8.

Reply:
Colorchange works, it's just been moved to a mode. That means that once you're in that mode, you are essentially isolated from your prop file controls until you exit, which is nice because you can avoid interference with other things in your prop without having to guard everything.  The mode has its own set of button controls, and the default are click POW to save, double click POW to cancel. You can customize that if you want. I suppose you could think of modes as sort of a mini prop file.

Reply:
> Colorchange works, it's just been moved to a mode....
Aha! I wasn't aware of that. Yes, just tried it and all good. 👍